server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -102,12 +102,21 @@ public static SqlSegmentsMetadataQuery forHandle(
    *
    * Returns a closeable iterator. You should close it when you are done.
    */
+  public CloseableIterator<DataSegment> retrieveUsedSegments(
+      final String dataSource,
+      final Collection<Interval> intervals,
```

Comment:
Since it is a new method, we might as well just pass a single Interval here rather than a collection.